EECORE-1487 bug with discussion forum

bakin1999 opened the pull request:
<!--
ExpressionEngine uses semantic versioning.

- (x.x.X) Bug fixes should target the stability branch
- (x.X.x) Small additive changes should target the next minor branch (release/next-minor if a numbered branch does not yet exist)
- (X.x.x) Breaking or large changes should target the next major branch (release/next-major if a numbered branch does not yet exist)
-->

<!-- What's in this pull request? -->


Replace this paragraph with a description of your changes and reasoning. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves a project issue, provide a link: -->


## Nature of This Change

<!-- Check all that apply: -->

- [x] 🐛 Fixes a bug
- [ ] 🚀 Implements a new feature
- [ ] 🛁 Refactors existing code
- [ ] 💅 Fixes coding style
- [ ] ✅ Adds tests
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security <!-- if your fix would EXPOSE a current security flaw, do not submit a pull request. Instead report a security bug at https://docs.expressionengine.com/latest/bugs_and_security_reports -->

## Is this backwards compatible?

- [x] Yes
- [ ] No

## Documentation
<!-- Required for new features -->
User Guide Pull Request: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/NNN

<!-- Don't forget to add a single line changelog for your change to the appropriate file!

- changelogs/patch.rst (x.x.X)
- changelogs/minor.rst (x.X.x)
- changelogs/major.rst (X.x.x)

Thank you for contributing to ExpressionEngine! -->

xhezairbey committed:
Fixed Files filters

- $reset_url breaks sorting filters when perPage is first selected
- Removed duplicate attribute where ‘list’ is the default
- Changed order of page vars initialization, since list is the default

xhezairbey committed:
ViewType filter cleanup.

Fixed class name, removed unused properties & refactored methods.

xhezairbey committed:
Added changelog line

robsonsobral committed:
Add `:trim` variable modifier

robsonsobral committed:
Add `:trim` variable modifier

robsonsobral committed:
Allow static `exp:category_heading` by adding `category_url_title` parameter

intoeetive committed:
cover static category headings with tests

robsonsobral committed:
Ignore VS Code workspace

robsonsobral committed:
Fix category identification

jcogs-design committed:
Modify template_post_parse hook per issue #1195
Write current template name / group to array variable
Add conditional to return WIP template when not final

jcogs-design committed:
Merge branch '6.dev' of https://github.com/jcogs-design/ExpressionEngine into 6.dev

jcogs-design committed:
Remove redundant code
Original idea was to use count($this->templates_loaded)
to provide access to current templates, but actually they
appear via $template / $template_group anyhow...

jcogs-design committed:
Issue #1195
Update patch changelog

jcogs-design committed:
Issue #1195 - Improved code
Hopefully this addresses issues flagged...
- PSR12 { position (corrected)
- if `ee('LivePreview')->hasEntryData()` missing (put back)
- code compacted somewhat
- conditional updated

jcogs-design committed:
Issue #1195 - fix typo
Final conditional was inverted - corrected.

robsonsobral committed:
Add tests

robsonsobral committed:
Add tests

intoeetive committed:
Merge branch 'feature/trim-variable-modifier' of github.com:robsonsobral/ExpressionEngine into feature/trim-variable-modifier

# Conflicts:
#	system/ee/ExpressionEngine/Service/Formatter/Formats/Text.php
#	tests/cypress/cypress/integration/field/modifiers.ee6.js
#	tests/cypress/support/templates/modifiers.group/layout-for-trim.html
#	tests/cypress/support/templates/modifiers.group/trim.html

robsonsobral committed:
Add channel data to category archive

intoeetive committed:
enable updating versioning setting for entries when channel setting is updated

added ChannelEntry service

fixing #367

EECORE-1237

robsonsobral committed:
Update changelog

robsonsobral committed:
Respect PSR-12

robsonsobral committed:
Respect PSR-12 (even more)

robsonsobral committed:
Respect PSR-12

intoeetive committed:
add tests for updating channel versioning settings

intoeetive committed:
fix typo in tests

robinsowell committed:
Add drop column batch method, fixes row size error in updates

intoeetive committed:
add entry_id and url_title as searchable fields on CP Entries listing page

EECORE-1331

intoeetive committed:
only preparse {member_search_path} if it's present in template

intoeetive committed:
do not join custom fields if they are disabled with tag param

if {exp:channel:entries} tag has disable="custom_fields" parameter,
we do not need to include custom fields into SQL query

also reduce fields chunk size for joins

fixes #1280

EECORE-1332

intoeetive committed:
cache member permissions directly in session library

intoeetive committed:
add sticky="none" parameter for exp:channel:entries

fixes #488

EECORE-1334

intoeetive committed:
allow modules to be used in menu manager

fixes #54

EECORE-1339

robsonsobral committed:
Add sticky entry column to entries table

intoeetive committed:
modification to comments UI

added more links between comment page and Comments Jump menu

EECORE-1129

intoeetive committed:
update text for Manu Manager jump menu item

jcogs-design committed:
Merge branch 'ExpressionEngine:6.dev' into Issue-no-1195

robinsowell committed:
Cleaning some comments and white space

robsonsobral committed:
Allow assign previously created field groups to importing channel sets

robsonsobral committed:
Unset loop vars out of the loop itself

robsonsobral committed:
Remove debug code

ajohnson committed:
Comments author info styling

intoeetive committed:
Merge branch 'bug/6.x/fix-the-fails' into feature/6.x/assign-role-on-signup

intoeetive committed:
add tests around member registration to different roles

intoeetive committed:
Merge branch '6.dev' into feature/6.x/assign-role-on-signup

# Conflicts:
#	system/ee/ExpressionEngine/Addons/member/mod.member_register.php
#	system/ee/ExpressionEngine/Controller/Members/Members.php
#	system/ee/language/english/members_lang.php
#	tests/cypress/cypress/integration/members/register.ee6.js

intoeetive committed:
add templates folder back

intoeetive committed:
fix the tests to run on GitHub Actions

intoeetive committed:
Merge branch '6.dev' into feature/6.x/assign-role-on-signup

intoeetive committed:
update cypress to run without warnings

intoeetive committed:
update comments widget

EECORE-1129

robsonsobral committed:
Sort sites menu

intoeetive committed:
Merge remote-tracking branch 'origin/HEAD' into feature/6.x/assign-role-on-signup

# Conflicts:
#	system/ee/ExpressionEngine/Tests/bootstrap.php
#	system/ee/installer/controllers/wizard.php
#	system/ee/legacy/libraries/Core.php

intoeetive committed:
Merge branch '6.dev' into feature/6.x/comments-page-ui

# Conflicts:
#	themes/ee/cp/css/common.min.css

intoeetive committed:
Merge branch '6.dev' into feature/trim-variable-modifier

# Conflicts:
#	tests/cypress/cypress/integration/field/modifiers.ee6.js

intoeetive committed:
Merge branch '6.dev' into feature/static-category-heading

intoeetive committed:
Merge branch '6.x' into feature/trim-variable-modifier

intoeetive committed:
Merge branch '6.dev' into feature/6.x/comments-page-ui

# Conflicts:
#	themes/ee/cp/css/common.min.css

intoeetive committed:
fix the issue where using exact keyword search on exp:channel:entries was causing PHP notices in Live Preview

fixes #1434

EECORE-1468

intoeetive committed:
Merge branch '6.dev' into feature/static-category-heading

intoeetive committed:
Merge branch '6.dev' into feature/6.x/comments-page-ui

intoeetive committed:
fix the trails for outgoing email prefs in jump menu

intoeetive committed:
do not show comment form if member role has no permission for it

EECORE-1471

robsonsobral committed:
Move sorting to right after the definition of list of sites

bakin1999 committed:
EECORE-1487 bug with discussion forum

robsonsobral committed:
Fix regression by 203b551c98917782351408fb5330a2ac7bfca5e7

intoeetive committed:
cookies registry modifications

register CP cookies separately
only register cookies if add-on is installed
add tests

EECORE-1490

intoeetive committed:
allow add-ons in custom menu even if there are no extensions

EECORE-1028

intoeetive committed:
add info to developer log when getting multibyte-related SQL error

and also silently continue when that happens during template synchrnozation

EECORE-1405

intoeetive committed:
Revert "add info to developer log when getting multibyte-related SQL error"

This reverts commit 36985b8a844621084ad06d9c1cb19345c65555d2.

intoeetive committed:
add info to developer log when getting multibyte-related SQL error

and also silently continue when that happens during template synchrnozation

EECORE-1405

intoeetive committed:
prefer preview based on Pages/channel settings
for edits triggered from Entries prolet

EEPRO-295

ops-andy commented:
@intoeetive please attach this to a milestone and update the corresponding Jira ticket. thanks!

ops-andy commented:
@intoeetive please attach this to a milestone and update the corresponding Jira ticket. thanks!

intoeetive reviewed (dismissed):
@bakin1999 I'm slightly unsure about usage of `300` number here - maybe if we have a constant, better to use it. Otherwise, approving

matthewjohns0n commented on system/ee/ExpressionEngine/Addons/forum/mod.forum_core.php:
```diff
@@ -339,14 +339,24 @@ public function _fetch_read_topics($new_id = false)
+            read_topics should be in the form of a topic id and a time int.
+            Value should also be recent in case someone is trying to put in incorrect time values (No future values allowed either)*/
+            $clean_read_topics = array();
+            foreach($read_topics as $key => $val){
+                if(is_int($val) && (ee()->localize->now - $val) < 300 && (ee()->localize->now - $val) > -1){
```
@bakin1999 I'm afraid this might introduce a bug. I'm thinking we should probably reduce this down to just the `is_int($val)` part. 

Also, I was hoping you could update the `foreach` to follow PSR-12.

bakin1999 committed:
Remove time checking

robsonsobral committed:
Parse conditionals on Request module...

...and also fix tagdata being removed instead of returned

TomJaeger committed:
Merge pull request #1534 from ExpressionEngine/feature/6.x/cookies-tag

cookies registry modifications

TomJaeger committed:
removed default sticky column.. renamed col from make entry sticky to Sticky Entry

TomJaeger committed:
Merge pull request #1306 from robsonsobral/feature/sticky_entry_column

Add sticky entry column to entries table

intoeetive committed:
update test process to build EE without Pro for third-party PRs

intoeetive committed:
Merge branch '6.dev' into feature/channel-data-on-category-archive

intoeetive committed:
add more tests for category heading

refactored code to use parse_variables_row
fix the bug with category image not parsed in nested mode

intoeetive committed:
fixed issue when templates with duplicate names from MSM sites were not synchronized

EECORE-1495

fixes #1509

matthewjohns0n committed:
PSR-12 Form field

matthewjohns0n committed:
Only include label span if the field label is set

Fixes #1569
EECORE-1537

matthewjohns0n committed:
Make the addon namespace generator more robust

EECORE-1421

intoeetive committed:
define `PATH_PRO_ADDONS` on installer

robsonsobral committed:
Fix wrong Last-Modified header for CSS requests

The header is being get from database row no matter if the template is on database or file

intoeetive committed:
skip building Pro for third-party PRs

intoeetive committed:
warn about wrong cookie domain

EECORE-1466

intoeetive committed:
rename the files that were triggering 406 error with mod_security

EECORE-1492

fixes #1491

intoeetive committed:
wait a minute!

intoeetive committed:
fixed the bug when non-superadmins could not upload avatars

fixes #1588

EECORE-1547

Yulyaswan committed:
fixed select box position in the modal window, grid field

intoeetive committed:
run the tests on heavy load and expect them to fail

intoeetive committed:
testing 8.1 support was too optimiztic here

intoeetive committed:
update the correct test

intoeetive committed:
the seed should succeed

intoeetive committed:
solve the issues with some CP parts not loading when too many members are present

add `total_members` property for Roles
add `ignore_member_stats` config override

fix display of selected author when a lot of authors exist in system
fixes #1582
EECORE-1548

fix issue with Communicate page not loading with many members being present
EECORE-1513
fixes #1533

display correct number of members for each role
EECORE-1516
fixes #1467

intoeetive committed:
update the installer code

intoeetive committed:
make the tests pass... maybe

intoeetive committed:
fix the issue when `{member_profile_link}` variable could be incorrect

fixes #1604
EECORE-1557

intoeetive committed:
use referenenced Pro branch for testing

intoeetive committed:
remove memory limit on tests

intoeetive committed:
split the test into parts

intoeetive committed:
reorganize the tests

intoeetive committed:
update the tests for the warnings

intoeetive committed:
add some waiting on tests

obfuscode committed:
Merge pull request #1386 from ExpressionEngine/feature/6.x/assign-role-on-signup

obfuscode committed:
PSR-12 fixes for PR

obfuscode committed:
Merge pull request #1338 from ExpressionEngine/bug/6.dev/drop-column-row-size

obfuscode committed:
Merge pull request #1299 from ExpressionEngine/feature/6.x/sticky-none

obfuscode committed:
Merge pull request #1291 from ExpressionEngine/feature/6.dev/search-by-entry-id

obfuscode committed:
Merge pull request #1239 from ExpressionEngine/feature/6.x/entry-revision-update

obfuscode committed:
Merge pull request #1292 from ExpressionEngine/feature/6.x/channel-entries-queries-optimization

obfuscode committed:
Merge pull request #1598 from ExpressionEngine/bug/6.x/column-type-selectbox-issue

obfuscode committed:
Merge pull request #1335 from jcogs-design/Issue-no-1195

TomJaeger committed:
Merge pull request #1593 from ExpressionEngine/feature/6.x/cookie-domain

warn about wrong cookie domain

obfuscode committed:
Merge pull request #1594 from ExpressionEngine/bug/6.x/mod-security-fix

obfuscode committed:
Merge branch '6.2.0' into feature/importing-channel-sets

obfuscode committed:
Merge pull request #1340 from robsonsobral/feature/importing-channel-sets

TomJaeger committed:
Merge pull request #1579 from robsonsobral/patch-3

Fix wrong Last-Modified header for CSS requests

obfuscode committed:
Merge branch '6.2.0' into feature/trim-variable-modifier

TomJaeger committed:
Merge pull request #1571 from ExpressionEngine/bug/6.dev/fix-cli-namespace

Make the addon namespace generator more robust

obfuscode committed:
Merge pull request #1183 from robsonsobral/feature/trim-variable-modifier

TomJaeger committed:
Merge pull request #1566 from ExpressionEngine/bug/6.x/template-sync-msm-issue

MSM template synchronization fix

obfuscode committed:
Merge branch '6.2.0' into feature/6.x/comments-page-ui

obfuscode committed:
Merge pull request #1328 from ExpressionEngine/feature/6.x/comments-page-ui

obfuscode committed:
Merge pull request #1308 from ExpressionEngine/feature/6.x/menu-manager

TomJaeger committed:
Merge pull request #1597 from ExpressionEngine/bug/6.x/avatar-uploads

fixed the bug when non-superadmins could not upload avatars

obfuscode committed:
Merge pull request #1489 from ExpressionEngine/bug/6.x/live-preview-notices

TomJaeger committed:
Merge pull request #1570 from ExpressionEngine/bug/6.dev/shared-view-form-short-text-value-required

Bug/6.dev/shared view form short text value required

obfuscode commented:
@bakin1999 It looks like there are a few failing tests for this. Any chance you can determine why or reach out to Yuri if you need assistance?

obfuscode reviewed (changes requested):
Failing tests need review.

TomJaeger committed:
Merge pull request #1538 from ExpressionEngine/feature/6.x/entries-prolet-preview-url

Pro Entries prolet: prefer preview based on Pages/channel settings

obfuscode committed:
Merge branch '6.2.0' into feature/static-category-heading

TomJaeger committed:
Merge pull request #1355 from robsonsobral/patch-1

Sort sites menu

obfuscode committed:
Merge pull request #1179 from robsonsobral/feature/static-category-heading

obfuscode committed:
Merge pull request #1535 from ExpressionEngine/bug/6.x/multibyte-errors

intoeetive committed:
Merge branch '6.2.0' into feature/conditionals-on-request-module

intoeetive committed:
Merge branch '6.2.0' into feature/channel-data-on-category-archive

intoeetive committed:
remove duplicate code

TomJaeger committed:
Merge pull request #1235 from robsonsobral/feature/channel-data-on-category-archive

Add channel data to category archive

TomJaeger committed:
Merge pull request #1576 from ExpressionEngine/bug/6.x/pro-addons-installer-error

define `PATH_PRO_ADDONS` on installer

TomJaeger committed:
Merge pull request #1605 from ExpressionEngine/bug/6.x/member_profile_link

fix the issue when `{member_profile_link}` variable could be incorrect

TomJaeger committed:
Merge pull request #1599 from ExpressionEngine/feature/6.x/members-heavy-load

add counter for roles

TomJaeger committed:
Merge branch '6.2.0' into feature/conditionals-on-request-module

TomJaeger committed:
Merge pull request #1562 from robsonsobral/feature/conditionals-on-request-module

Parse conditionals on Request module...

intoeetive committed:
re-enable some PHPUnit tests for PHP8

EECORE-888

intoeetive committed:
update test environment

obfuscode committed:
Merge pull request #1621 from ExpressionEngine/feature/6.x/php-8-unit-tests

obfuscode committed:
Merge branch '6.2.0' into bug/6.x/files-perpage-resets-filters

obfuscode committed:
Merge pull request #1116 from xhezairbey/bug/6.x/files-perpage-resets-filters

obfuscode commented:
@bakin1999 Can you merge in the latest 6.2.0 branch so we can re-run these tests.

bakin1999 commented:
ok, am changing to draft then opening again to re-trigger tests

bakin1999 committed:
Merge remote-tracking branch 'upstream/6.2.0' into bugg/EECORE-1487
merge

intoeetive reviewed (changes requested):
It is currently showing 122 changed files, so I have no idea what this change is about. Probably needs to be rebased
